gateway_rejection_reason.go

```diff
@@ -0,0 +1,14 @@
+package braintree
+
+type GatewayRejectionReason string
+
+const (
+	GatewayRejectionReasonApplicationIncomplete GatewayRejectionReason = "application_incomplete"
+	GatewayRejectionReasonAVS                                          = "avs"
+	GatewayRejectionReasonAVSAndCVV                                    = "avs_and_cvv"
+	GatewayRejectionReasonCVV                                          = "cvv"
+	GatewayRejectionReasonDuplicate                                    = "duplicate"
+	GatewayRejectionReasonFraud                                        = "fraud"
+	GatewayRejectionReasonThreeDSecure                                 = "three_d_secure"
+	GatewayRejectionReasonUnrecognized                                 = "unrecognized"
```

Comment:
Nice catch! 

FWIW, I looked around to see what the current state of things was to mimic it-- do you think it's worth a potentially breaking change to go around and update those for consistency? main.go, for example, has the bug you pointed out here (only one of the constants is typed). subscription.go has the same convention format as this, but has a string type. That kind of thing. disbursement.go does not have the same naming convention as this and is untyped.